convert2rhel/pkghandler.py

```diff
@@ -175,30 +179,54 @@ def get_installed_pkgs_w_fingerprints(name=""):
     package_objects = get_installed_pkg_objects(name)
     pkgs_w_fingerprints = []
     for pkg_obj in package_objects:
-        pkg_sig = pkg_obj.hdr.sprintf("%|DSAHEADER?{%{DSAHEADER:pgpsig}}:"
-                                      "{%|RSAHEADER?{%{RSAHEADER:pgpsig}}:"
-                                      "{(none)}|}|")
-        fingerprint_match = re.search("Key ID (.*)", pkg_sig)
-        if fingerprint_match:
-            pkgs_w_fingerprints.append(PkgWFingerprint(
-                pkg_obj, fingerprint_match.group(1)))
+        if pkgmanager.TYPE == 'yum':
+            pkg_sig = pkg_obj.hdr.sprintf("%|DSAHEADER?{%{DSAHEADER:pgpsig}}:"
+                                          "{%|RSAHEADER?{%{RSAHEADER:pgpsig}}:"
+                                          "{(none)}|}|")
+            fingerprint_match = re.search("Key ID (.*)", pkg_sig)
+            if fingerprint_match:
+                pkgs_w_fingerprints.append(PkgWFingerprint(
+                    pkg_obj, fingerprint_match.group(1)))
+            else:
+                pkgs_w_fingerprints.append(PkgWFingerprint(pkg_obj, "none"))
         else:
-            pkgs_w_fingerprints.append(PkgWFingerprint(pkg_obj, "none"))
+            dnf_base = pkgmanager.Base()
+            dnf_base.fill_sack(load_system_repo=True, load_available_repos=False)
+            try:
+                sig_check, _ = dnf_base._sig_check_pkg(pkg_obj)
+            except KeyError:
+                sig_check = 1
+            if sig_check == 0: # OK
+                pkgs_w_fingerprints.append(PkgWFingerprint(pkg_obj, pkg_obj.reponame))
+            else:
+                pkgs_w_fingerprints.append(PkgWFingerprint(pkg_obj, "none"))
     return pkgs_w_fingerprints
 
 
 def get_installed_pkg_objects(name=""):
     """Return list with installed package objects. The packages can be
     optionally filtered by name.
     """
-    yum_base = yum.YumBase()
+    if pkgmanager.TYPE == 'yum':
+        return _get_installed_pkg_objects_yum(name)
+    return _get_installed_pkg_objects_dnf(name)
+
+def _get_installed_pkg_objects_yum(name):
+    yum_base = pkgmanager.YumBase()
     # Disable plugins (when kept enabled yum outputs useless text every call)
     yum_base.doConfigSetup(init_plugins=False)
     if name:
         return yum_base.rpmdb.returnPackages(patterns=[name])
-
     return yum_base.rpmdb.returnPackages()
 
+def _get_installed_pkg_objects_dnf(name):
+    dnf_base = pkgmanager.Base()
+    dnf_base.fill_sack(load_system_repo=True, load_available_repos=False)
+    query = dnf_base.sack.query()
+    installed = query.installed()
+    if name:
+        installed.filter(name=name)
+    return list(installed)
```

Comment:
These `*_(yum|dnf)` methods should be moved into the pkgmanager module. What I expected to see from my feedback is that all the gunk you're doing to do either YUM or DNF would be shunted into classes for YUM and DNF variants of the function, and then the pkghandler module would instantiate the YUM or DNF variant depending on import availability (preferring DNF, obviously).

Reply:
I totally agree and I've started working on that but if you don't mind, I'd actually submit that as another PR. This one is huge enough on its own now and this additional work requires quite a lot of refactoring (I'd move all the yum/dnf-related code to the pkgmanager.py module, and there's quite a lot of it).

Reply:
I'm fine with it being a followup PR, but I think you should not make a release until after that change is done.

Reply:
Agreed.